lumina-inc/lumina-lambda
# Patch license key generation & login

## Proposed Changes

- Modify character combination of license key from `3 characters of last name + customer id` to `last 3 characters of phone number + customer id` when generating license key
- Modify login end-point so that it does not validate the first 3 characters of decrypted license key with the first 3 characters of customer's `last name`

## Expected Results

- By hitting account activation end-point, you can generate license key & send customer activation email (even with non-ASCII characters, for example `Djodi 千尋`)
- By hitting login end-point, you can login with the old & new combination of license key

## How to Test

- Create a new customer on Shopify. Make sure you fill in `Customer Overview` section and `Address` section. Only need Phone number in `Address` section
<img width="614" alt="Screen Shot 2022-01-07 at 13 49 49" src="https://user-images.githubusercontent.com/36635194/148504096-f2da0db2-1566-4d23-9554-7a1c9ce55652.png">
<img width="611" alt="Screen Shot 2022-01-07 at 13 49 57" src="https://user-images.githubusercontent.com/36635194/148504087-da8bdbd4-2e21-44bb-a0a9-4abf6348b771.png">

- Activate account using received email

- Use license key to hit login end-point. You can use documentation below, or [https://documenter.getpostman.com/view/18813947/UVR8oSxV](url)
```
POST: https://lumina-lambda-staging.netlify.app/.netlify/functions/login
Headers: Content-Type: application/json
Body: 
{
    "license_key":"39B8CF7B-945D38A7-68E9F8DB-75489A2D"
}
```

- You can also use license key that is generated before today (2021-01-07) as well. For example `39B8CF7B-945D38A7-68E9F8DB-75489A2D`
